python/paddle/fluid/dygraph/amp/auto_cast.py

```diff
@@ -0,0 +1,166 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
```

Comment:
auto_cast.py and loss_scaller.py have many duplicate codes with mixed precision in the static graph.

Reply:
And should they need to adjust as the same interface?

Reply:
We will consider to unify dygraph API and Static API of AMP in future.